shared/js/background/classes/ad-click-attribution-policy.js

```diff
@@ -134,9 +157,26 @@ export class AdClick {
         restoredAdClick.adClickRedirect = adClick.adClickRedirect
         restoredAdClick.expires = adClick.expires
         restoredAdClick.clickExpires = adClick.clickExpires
+        restoredAdClick.adClickDNR = adClick.adClickDNR
         return restoredAdClick
     }
 
+    /**
+     * @param {string} domain
+     * @param {boolean} adClickRedirect
+     **/
+    setAdBaseDomain (domain, adClickRedirect) {
+        this.adBaseDomain = domain
+        this.adClickRedirect = adClickRedirect
+        this.adClickDNR?.updateInitiator(domain)
+    }
+
+    removeAdClickDNR () {
+        if (this.adClickDNR) {
+            chrome.declarativeNetRequest.updateSessionRules({ removeRuleIds: [this.adClickDNR.rule.id] })
```

Comment:
This either belongs in AdClickDNR or AdClick should take care of all AdClickDNR's code too.

So either:
```suggestion
           this.adClickDNR?.remove()
```

Or bring all the class in here :)

Reply:
If the former, it totally could be called from within hasNotExpired

Reply:
I'll just move the class in here.  Calling remove on adClickDNR would require recreating it when cloning which gets complicated.